src/python/pants/backend/shell/target_types.py

```diff
@@ -46,6 +42,20 @@ class ShellGeneratingSourcesBase(MultipleSourcesField):
     uses_source_roots = False
 
 
+class GeneratorSettingsRequest(TargetFilesGeneratorSettingsRequest):
+    pass
```

Comment:
Right, so maybe I left my review for these changes on the wrong PR... 😅 

I find this name too generic, how about `ShellGeneratorSettingsRequest` ?